src/classes/RD2_ValidationService.cls

```diff
@@ -280,6 +281,22 @@ public with sharing class RD2_ValidationService {
             && oldAcct.npe01__SYSTEMIsIndividual__c == false;
     }
 
+    /***
+    * @description Return true when account is changed from null to contact's account and contact is not changed.
+    * @param new recurring donation record
+    * @param old recurring donation record
+    * @return Boolean True when account is updated to contact's account from null value
+    */
+    private Boolean isAccountUpdatedToContactAccount(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd) {
```

Comment:
The following method can be replaced with:
```
/***
    * @description Returns true when Account on RD is changed from null to the Contact's Account
    * and the Contact on RD is not changed
    * @param new Updated Recurring Donation record
    * @param old Recurring Donation record containing old values
    * @return Boolean
    */
    private Boolean isAccountAutoPopulatedOnChange(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd) {

        if (rd.npe03__Contact__c == null //The Contact is not specified
            || rd.npe03__Contact__c != oldRd.npe03__Contact__c //The Contact is changed on update
            || oldRd.npe03__Organization__c != null //The Account existed previously
        ) {
            return false;
        }
        
        Contact rdContact = contactById.get(rd.npe03__Contact__c);

        return rdContact != null
            && rd.npe03__Organization__c == rdContact.AccountId;
    }
```

where the following can be added on line 80:
```

    /**
    * @description Contacts on provided Recurring Donations
    */
    private Map<Id, Contact> contactById {
        get {
            if (contactById == null) {
                Set<Id> contactIds = UTIL_SObject.extractId(
                    rds, npe03__Recurring_Donation__c.npe03__Contact__c
                );

                contactById = new Map<Id, Contact>([
                    SELECT Name, AccountId
                    FROM Contact
                    WHERE Id IN :contactIds
                ]);
            }
            return contactById;
        } 
        set;
    }
```